File: arrays/binary_search.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function binarySearch(haystack, needle) {
    var start = 0;
    var end = haystack.length;
    var midpoint, value;
    do {
        midpoint = Math.floor((start + end) / 2);
        value = haystack[midpoint];
        if (value === needle) {
            return true;
        }
        else if (value > needle) {
            end = midpoint;
        }
        else {
            start = midpoint + 1;
        }
    } while (start < end);
    return false;
}
exports.default = binarySearch;
console.log(binarySearch([1, 2, 3, 4, 5], 3));

File: typescript/arrays/binary_search.py
from math import floor
from typing import List


def binary_search(haystack: List[int], needle: int):
    start = 0
    end = len(haystack)

    while start < end:
        midpoint = floor((start + end) / 2)
        value = haystack[midpoint]

        if value == needle:
            return True
        elif value > needle:
            end = midpoint
        else:
            start = midpoint + 1

    return False


print(binary_search([1, 2, 3, 4, 5], 6))
